### PR TITLE
Fix parse inconsistency in left shift

### LINKED ISSUE #1143: Parse inconsistency in left shift
Currently, we can have a line break following all other binary operators (including the right shift) except the left shift. Following code will result in an index out-of-range error, whereas it will work with any other operator such as `+` or `>>`
```
import ballerina/io;
public function main() {
    int a = 5;
    int b = 10;
    int c = a <<
    b;
    io:println(c);
}

```

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Fix parse inconsistency in the left shift
Resolves #1143 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/modules/front.syntax/scan.bal [hunk 1]
```diff
@@ -457,7 +457,7 @@ function scanNormal(int[] codePoints, int startIndex, Scanned result) {
                     int cp2 = codePoints[i];
                     if cp2 == CP_LESS_THAN {
                         i += 1;
-                        if codePoints[i] == CP_EQUAL {
+                        if i < len && codePoints[i] == CP_EQUAL {
                             i+=1;
```
Comment (jclark) [inline compiler/modules/front.syntax/scan.bal]: Space around +=

**Review by jclark — CHANGES_REQUESTED** [review]
Add parsing specific test case

File: compiler/modules/front.syntax/scan.bal (hunk 1 above)
Comment (heshanpadmasiri) [reply compiler/modules/front.syntax/scan.bal]: fixed

**Comment by heshanpadmasiri** [comment]
> Add parsing specific test case

fixed

**Review by jclark — APPROVED** [review]

**Pull request merged**